### PR TITLE
Add nightly and weekly whl tests

### PR DESCRIPTION
**Pull request by mengfei25**

### CODE REVIEW

File: .github/workflows/nightly_ondemand_whl.yml [hunk 1]
```diff
@@ -0,0 +1,365 @@
+name: Nightly-OnDemand WHL Tests
```
Comment (chuanqi129) [inline .github/workflows/nightly_ondemand_whl.yml:1]: Let's rename it as `Torch Nightly WHL Tests`

File: .github/workflows/nightly_ondemand_whl.yml [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+          echo "TIMM_COMMIT_ID=$(<.ci/docker/ci_commit_pins/timm.txt)" |tee -a "${GITHUB_OUTPUT}" >> "${GITHUB_ENV}"
+          echo "MODEL_ONLY_NAME=${{ inputs.model }}" |tee -a "${GITHUB_OUTPUT}" >> "${GITHUB_ENV}"
+          source /opt/intel/oneapi/pytorch-gpu-dev-0.5/oneapi-vars.sh
+          echo "DRIVER_VERSION=$(dkms status 2>&1 |grep 'intel-i915-dkms' |sed 's/.*\///;s/,.*//')" |tee -a "${GITHUB_OUTPUT}" >> "${GITHUB_ENV}"
+          echo "BUNDLE_VERSION=$(dpcpp --version 2>&1 |grep 'DPC++/C++' |sed 's/.*(//;s/).*//')" |tee -a "${GITHUB_OUTPUT}" >> "${GITHUB_ENV}"
+          . /etc/os-release
+          echo "OS_PRETTY_NAME=${PRETTY_NAME}" |tee -a "${GITHUB_OUTPUT}" >> "${GITHUB_ENV}"
+          echo "GCC_VERSION=$(gcc -dumpversion)" |tee -a "${GITHUB_OUTPUT}" >> "${GITHUB_ENV}"
+          echo ${GITHUB_ENV}
+      - name: Triton Installation
```
Comment (chuanqi129) [inline .github/workflows/nightly_ondemand_whl.yml:155]: Don't need this step anymore. Triton will be installed by default as a dependency of torch nightly wheel

File: .github/workflows/nightly_ondemand_whl.yml (hunk 1 above)
Comment (mengfei25) [reply .github/workflows/nightly_ondemand_whl.yml:1]: Do you mean we just need keep nightly and remove all on-demand?

File: .github/workflows/nightly_ondemand_whl.yml (hunk 2 above)
Comment (mengfei25) [reply .github/workflows/nightly_ondemand_whl.yml:155]: okay, will remove the useless code

File: .github/workflows/nightly_ondemand_whl.yml [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+          echo "OS_PRETTY_NAME=${PRETTY_NAME}" |tee -a "${GITHUB_OUTPUT}" >> "${GITHUB_ENV}"
+          echo "GCC_VERSION=$(gcc -dumpversion)" |tee -a "${GITHUB_OUTPUT}" >> "${GITHUB_ENV}"
+          echo ${GITHUB_ENV}
+      - name: Build Pytorch XPU
+        run: |
+          source activate e2e_ci
+          source .github/scripts/env.sh
+          cd ../pytorch
+          pip install -r requirements.txt
+          export CMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}:${CONDA_PREFIX:-"$(dirname $(which conda))/../"}
```
Comment (chuanqi129) [inline .github/workflows/nightly_ondemand_whl.yml:141]: don't need this anymore, change the step name to install pytorch xpu nightly whl

File: .github/workflows/nightly_ondemand_whl.yml [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+          pwd
+          cd ../ && rm -rf pytorch
+          source activate e2e_ci
+          git clone https://github.com/pytorch/pytorch pytorch
+          cd pytorch && git checkout ${{ env.pytorch }} 
+          # apply PRs for stock pytorch
+          pip install requests
+          python ../torch-xpu-ops/.github/scripts/apply_torch_pr.py
+          git status && git show -s
+      - name: Identify pinned versions
```
Comment (chuanqi129) [inline .github/workflows/nightly_ondemand_whl.yml:114]: may need to consider how to get accurate sw info in this whl test workflow, eg. torch/torchvision/audio version, we may get it from pip list after installation

File: .github/workflows/_linux_ut.yml [hunk 5]
```diff
@@ -79,9 +83,13 @@ jobs:
...
           fi
-          WERROR=1 python setup.py bdist_wheel
-          pip install --force-reinstall dist/*.whl
-          git clone https://github.com/pytorch/vision && cd vision && python setup.py install && cd ..
+          if [ ${{ inputs.test_type }} != "whl" ]; then
+            WERROR=1 python setup.py bdist_wheel
+            pip install --force-reinstall dist/*.whl
+            git clone https://github.com/pytorch/vision && cd vision && python setup.py install && cd ..
+          else
+            pip install torch torchvision torchaudio pytorch-triton-xpu --pre --index-url https://download.pytorch.org/whl/nightly/xpu
```
Comment (chuanqi129) [inline .github/workflows/_linux_ut.yml:91]: let's keep standard nightky wheel installation cmd, remove the triton in the cmd, it's a dependency of torch.

File: .github/workflows/nightly_ondemand.yml [hunk 6]
```diff
@@ -352,9 +352,6 @@ jobs:
           echo -e "$RUNNER_NAME | $OS_PRETTY_NAME | $GCC_VERSION | ${{ env.python }} | $DRIVER_VERSION| $BUNDLE_VERSION \n" >> ${{ github.workspace }}/report.txt
           if [ "${GITHUB_EVENT_NAME}" == "workflow_dispatch" ];then
             test_scope="${{ inputs.suite }}/${{ inputs.dt }}/${{ inputs.mode }}/${{ inputs.scenario }}"
-            if [ "${{ inputs.triton }}" != "" ];then
```
Comment (chuanqi129) [inline .github/workflows/nightly_ondemand.yml:355]: what's the purpose of this change?

File: .github/workflows/nightly_ondemand_whl.yml (hunk 4 above)
Comment (mengfei25) [reply .github/workflows/nightly_ondemand_whl.yml:114]: Sure, torch/torchvision/torchaudio/triton versions are from pip list

File: .github/workflows/nightly_ondemand.yml (hunk 6 above)
Comment (mengfei25) [reply .github/workflows/nightly_ondemand.yml:355]: Triton version already in above table, no need again here

File: .github/workflows/nightly_ondemand_whl.yml [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+          echo "TRITON_COMMIT_ID=$(pip list |grep -w pytorch-triton-xpu |awk '{print $2}')" |tee -a "${GITHUB_OUTPUT}" >> "${GITHUB_ENV}"
+          echo "TORCH_BRANCH_ID=nightly" |tee -a "${GITHUB_OUTPUT}" >> "${GITHUB_ENV}"
+          echo "TORCH_COMMIT_ID=$(pip list |grep -w torch |awk '{print $2}')" |tee -a "${GITHUB_OUTPUT}" >> "${GITHUB_ENV}"
+          echo "TORCHBENCH_COMMIT_ID=$(<third_party/torch-xpu-ops/.github/ci_commit_pins/torchbench.txt)" |tee -a "${GITHUB_OUTPUT}" >> "${GITHUB_ENV}"
+          echo "TORCHVISION_COMMIT_ID=$(pip list |grep -w torchvision |awk '{print $2}')" |tee -a "${GITHUB_OUTPUT}" >> "${GITHUB_ENV}"
+          echo "TORCHAUDIO_COMMIT_ID=$(pip list |grep -w torchaudio |awk '{print $2}')" |tee -a "${GITHUB_OUTPUT}" >> "${GITHUB_ENV}"
+          echo "TRANSFORMERS_VERSION=$(<.ci/docker/ci_commit_pins/huggingface.txt)" |tee -a "${GITHUB_OUTPUT}" >> "${GITHUB_ENV}"
+          echo "TIMM_COMMIT_ID=$(<.ci/docker/ci_commit_pins/timm.txt)" |tee -a "${GITHUB_OUTPUT}" >> "${GITHUB_ENV}"
+          echo "MODEL_ONLY_NAME=${{ inputs.model }}" |tee -a "${GITHUB_OUTPUT}" >> "${GITHUB_ENV}"
+          source /opt/intel/oneapi/pytorch-gpu-dev-0.5/oneapi-vars.sh
```
[the remark this answers is not in the document]
Comment (mengfei25) [reply .github/workflows/nightly_ondemand_whl.yml:128]: collect bundle version infos and because you lost it here when changed to env.sh several days ago

**Review by chuanqi129 — APPROVED** [review]
Let's launch a on-demand test based on it, If it works fine, let's merge it

**Comment by mengfei25** [comment]
failed not related with this change
